internal/state/registry.go

```diff
@@ -48,6 +50,19 @@ func sFnRegistryConfiguration(ctx context.Context, r *reconciler, s *systemState
 	return nextState(sFnOptionalDependencies)
 }
 
+func setExternalRegistrySecretNameInDockerRegistryStatus(ctx context.Context, r *reconciler, s *systemState) {
+	// doc: https://kyma-project.io/docs/kyma/latest/05-technical-reference/svls-03-switching-registries#cluster-wide-external-registry
+	secret, err := registry.GetExternalRegistrySecret(ctx, r.client, s.instance.GetNamespace())
+	// ignore errors because it only set status
+	if err != nil || secret == nil {
+		return
+	}
+	if registryAddress, ok := secret.Data["registryAddress"]; ok {
+		s.instance.Status.DockerRegistry = string(registryAddress)
```

Comment:
I think we should use here the serverAddress field. In case when user uses, for example, Dockerhub, registryAddress contains the same value as the username field, and serverAddress reflects the real endpoint ( read the note [here](https://github.com/kyma-project/serverless-manager#using-serverless-manager) )